flink-dstl/flink-dstl-dfs/src/main/java/org/apache/flink/changelog/fs/FsStateChangelogWriter.java

```diff
@@ -302,14 +302,16 @@ public void reset(SequenceNumber from, SequenceNumber to) {
     }
 
     private static ChangelogStateHandleStreamImpl buildHandle(
-            KeyGroupRange keyGroupRange, NavigableMap<SequenceNumber, UploadResult> results) {
+            KeyGroupRange keyGroupRange,
+            NavigableMap<SequenceNumber, UploadResult> results,
+            long incrementalSize) {
         List<Tuple2<StreamStateHandle, Long>> tuples = new ArrayList<>();
         long size = 0;
         for (UploadResult uploadResult : results.values()) {
             tuples.add(Tuple2.of(uploadResult.getStreamStateHandle(), uploadResult.getOffset()));
             size += uploadResult.getSize();
         }
-        return new ChangelogStateHandleStreamImpl(tuples, keyGroupRange, size);
+        return new ChangelogStateHandleStreamImpl(tuples, keyGroupRange, size, incrementalSize);
```

Comment:
I'm afraid `incrementalSize` computed here doesn't take pre-emptively uploaded state into account; it only computes the size of the last portion before checkpoint.
And in most cases, most of the state should be pre-emptively uploaded.

Reply:
If we take the pre-emptively uploaded changelogs as checkpointed data size, the semantics would be broken as those changelogs does not uploaded in that checkpoint.

Reply:
I think you're right in the sense that pre-emptively uploaded changes do not affect sync/async phase.
But I'd say those changes **are** uploaded during that checkpoint.
I think confusion comes from the definition of "during that checkpoint". It can mean two things: 
1. the time from triggering a checkpoint to its completion (i.e. sync+async phase in task context)
2. the time between two barriers

I think we should explicitly clarify that we mean 1st one the docs and UI:
"during that checkpoint" -> "during sync and async phases of that checkpoint".
WDYT?

Reply:
I think it's fine to give such description in details. If so, I think we have reached the aggreement that we do not count pre-emptively uploaded changelogs as the checkpointed data size.